xds/src/main/java/io/grpc/xds/XdsServerWrapper.java

```diff
@@ -224,6 +224,7 @@ public void run() {
           delegate.shutdownNow();
         }
         internalShutdown();
+        initialStartFuture.set(new IOException("server is forcefully shut down"));
```

Comment:
The change looks good but is there any reason why we cannot extend it to `shutdown()` as well? The doc for `shutdown()` says this: "Initiates an orderly shutdown in which preexisting calls continue but new calls are rejected. After this call returns, this server has released the listening socket(s) and may be reused by another server... Calling this method before `start()` will shut down and terminate the server like normal...". Does not say anything about what happens if `shutdown()` is called *during* `start()` but it is reasonable to extrapolate that the effect should be similar to `shutdownNow()` since RPCs have not started. Just brainstorming about what's the right thing to do.

Reply:
Good question. Yes the javadoc does not say anything about what happens if `shutdown()` is called during `start()`. We can compare the following behaviors of `start()` for the situation.

- Return immediately. But according to the javadoc, after `start()` returns, clients may begin connecting to the listening socket(s). This seems not the case.
- Throw IllegalStateException. Kind of comply with the javadoc "throws IllegalStateException if already started or shut down" if we consider this situation as "already shut down"
- Throw IOException. Kind of comply with the javadoc "throws IOException if unable to bind", but an _orderly_ shutdown causes `start()` to throw.
- keep waiting. 

I can change to "throws IOException" (the same as `shutdownNow` case) if you do think it makes more sense.

Reply:
> * Throw IllegalStateException. Kind of comply with the javadoc "throws IllegalStateException if already started or shut down" if we consider this situation as "already shut down"

This probably comes closest although the state wasn't illegal when `start()` started and was put into the illegal state by a call to `shutdown()`.  We can also change the Java doc for `Server.start()` to expand the definition of IllegalState. But at the same time I don't have any strong opinions about this so this LGTM